data_diff/__init__.py

```diff
@@ -1,9 +1,86 @@
-from typing import Tuple
+from typing import Tuple, Iterator
 
 from .database import connect_to_uri
-from .diff_tables import TableSegment, TableDiffer
+from .diff_tables import (
+    TableSegment,
+    TableDiffer,
+    DEFAULT_BISECTION_THRESHOLD,
+    DEFAULT_BISECTION_FACTOR,
+    DbKey,
+    DbTime,
+    parse_table_name,
+)
 
 
-def create_source(db_uri: str, table_name: str, key_column: str, extra_columns: Tuple[str, ...] = ()):
-    db = connect_to_uri(db_uri)
-    return TableSegment(db, (table_name,), key_column, tuple(extra_columns))
+class TableRef:
+    """Connects to a URI and creates a table ref
+
+    This is an auxiliary class for diff_tables()
+    """
+
+    def __init__(self, db_uri: str, table_name: str):
+        self.db = connect_to_uri(db_uri)
+        self.table_name = table_name
+
+    def create_table_segment(self, **kwargs):
+        return TableSegment(self.db, parse_table_name(self.table_name), **kwargs)
+
+
+def diff_tables(
+    table1: TableRef,
+    table2: TableRef,
+    *,
+    # Name of the key column, which uniquely identifies each row (usually id)
+    key_column: str = "id",
```

Comment:
I've been thinking about whether we should call this the "bisection_column", but I think that might be leaking too many implementation details... Just important to note that this could be `tenant_id`, `created_at`, ... whatever. As long as it's numeric and has a decent distribution.

Reply:
created_at isn't guaranteed to be unique, so it won't work well as a key.

Reply:
Hmmm does the new algorithm care if they're unique though?

Reply:
I think the bisection will still work, as long as it's mostly unique. But it will mess with the stats feature.